src/dmd/backend/evalu8.d

```diff
@@ -172,9 +172,9 @@ version (SCPP)
 
                 case TYfloat4:
                 {   b = 0;
-                    for (size_t i = 0; i < 4; i++)
+                    foreach (f; e.EV.Vfloat4)
                     {
-                        if (isnan(e.EV.Vfloat4[i]) || e.EV.Vfloat4[i] != 0)
+                        if (isnan(f) || f != 0) // Why not just f != 0?
```

Comment:
It seems like `if (isnan(f) || f != 0)` could be just `if (f != 0)`.

Reply:
It's a relic from the days when it was in C and C compiler support for NaN was erratic and mostly wrong.